R/build-home-citation.R

```diff
@@ -6,10 +6,13 @@ has_citation <- function(path = ".") {
 create_meta <- function(path) {
   path <- path(path, "DESCRIPTION")
 
-  dcf <- read.dcf(path)
-  meta <- as.list(dcf[1, ])
+  dcf <- read.dcf(path)[1, ]
+  if ("Encoding" %in% names(dcf)) {
+    Encoding(dcf) <- dcf[["Encoding"]]
```

Comment:
I think my approach in https://github.com/r-lib/pkgdown/pull/679/files#diff-4603a117ddf509c48012889b8c3ad44bR12 is more correct. But why doesn't it work?

Reply:
You are probably right. Your PR to re-encode the dcf content is closer to what, e.g., `packageDescription()` does, so might be more correct. ;) It does fix the `citation(auto=meta)` issue. My PR avoids lapply and makes use of the efficient enc2utf8 primitive. But maybe there are drawbacks.